scipy/sparse/linalg/eigen/arpack/arpack.py

```diff
@@ -1864,13 +1865,22 @@ def matmat_XH_X(x):
 
         eigvals, eigvec = lobpcg(XH_X, X, tol=tol ** 2, maxiter=maxiter,
                                  largest=largest)
-
+    elif solver == 'propack':
+        from scipy.sparse.linalg import svdp
+        jobu = ((isinstance(return_singular_vectors, str) and 'u' in return_singular_vectors)
+                or (isinstance(return_singular_vectors, bool) and return_singular_vectors))
+        jobv = ((isinstance(return_singular_vectors, str) and 'v' in return_singular_vectors)
+                or (isinstance(return_singular_vectors, bool) and return_singular_vectors))
+        return svdp(
+            A, k=k, tol=tol**2, which=which, maxiter=maxiter, compute_u=jobu, compute_v=jobv,
```

Comment:
Do we know why `tol` is squared or whether that's appropriate for Propack, or is this assumed? If assumed, I should look into it more closely.

Is `maxiter` of PROPACK comparable to `maxiter` of ARPACK, considering that for PROPACK it's only used when `irl_mode` is True?

Reply:
This has been noted before (but not resolved) [here](https://github.com/scipy/scipy/pull/10830#pullrequestreview-293904906).

Reply:
I squared `tol` here because the other solvers had it squared and I didn't look into it, so might be worth thinking about a little more.

The `maxiter` shares a name but appears to be different than ARPACK's `maxiter`.  PROPACK references the number of restarts allowed in the IRL mode.